packages/ember-glimmer/lib/environment.js

```diff
@@ -275,16 +275,16 @@ export default class Environment extends GlimmerEnvironment {
     this.inTransaction = true;
 
     super.begin();
-
-    this.destroyedComponents = [];
   }
 
   commit() {
+    let destroyedComponents = this.destroyedComponents;
+    this.destroyedComponents = [];
```

Comment:
if the loop fails `destroy` could throw, we still likely want to release the contents of the original `destroyedComponents`